provider/ansible.py

```diff
@@ -149,7 +150,26 @@ def distro_install():
         """
         Install ansible from the distro
         """
-        return utils_package.package_install('ansible')
+        # Provide custom dnf repo containing ansible
+        install_status = False
+        if params.get("ansible_repo"):
+            repo_options = {
+                "priority": "1",
+                "gpgcheck": "0",
+                "skip_if_unavailable": "1"
+            }
+            yum_backend = YumBackend()
+            if yum_backend.add_repo(params["ansible_repo"], repo_options):
+                LOG_JOB.info(f"Ansible repo was added: {params['ansible_repo']}")
+            else:
+                LOG_JOB.error("Ansible repo was required, but failed to be added.")
+                return install_status
+        if not utils_package.package_install('ansible'):
+            LOG_JOB.error(f"Failed to install ansbile.")
+        else:
+            install_status = True
+        yum_backend.remove_repo(params["ansible_repo"])
+        return install_status
```

Comment:
What do you think about:
```
install_status = utils_package.package_install('ansible')
if not install_status:
    LOG_JOB.error("Failed to install ansible from distro")
yum_backend.remove_repo(params["ansible_repo"])
return  install_status
```

For L154 and L166,  it can use `return False` directly :)

Reply:
Yeah, I see I "overengineered" this part a bit :slightly_smiling_face: I agree, let's change it the way you proposed. (Done.)